pennylane/pulse/rydberg_hamiltonian.py

```diff
@@ -0,0 +1,237 @@
+
+
+class RydbergHamiltonian(ParametrizedHamiltonian):
+    r"""Class representing the Hamiltonian of an ensemble of Rydberg atoms under the action of
+    local and global laser fields:
+
+    .. math::
+
+        H = \hbar \frac{1}{2} \sum_i  \Omega_i(t) (\cos(\phi)\sigma_i^x - \sin(\phi)\sigma_i^y) -
+        \frac{1}{2} \sum_i \delta_i(t) \sigma_i^z + \sum_{i<j} V_{ij} n_i n_j
+
+    where :math:`\Omega_i` and :math:`\delta_i` correspond to the amplitude and detuning of the
+    laser applied to atom :math:`i`, and :math:`\sigma^\alpha` for :math:`\alpha = x,y,z` are
+    the Pauli matrices.
+
+    The last term of the sum corresponds to the interaction between the atoms of the ensemble
+    due to the Rydberg blockade. For more info, check :func:`rydberg_interaction`.
+
+    .. note::
+
+        :math:`\hbar` is set to 1.
+
+    .. warning::
+
+        When adding a ``RydbergHamiltonian`` with a :class:`ParametrizedHamiltonian` all the
+        information needed to translate this class into hardware will be lost.
+
+    .. seealso:: :class:`rydberg_interaction`, :class:`rydberg_transition`
+
+
+    Args:
+        coeffs (Union[float, callable]): coefficients of the Hamiltonian expression, which may be
+            constants or parametrized functions. All functions passed as ``coeffs`` must have two
+            arguments, the first one being the trainable parameters and the second one being time.
+        observables (Iterable[Observable]): observables in the Hamiltonian expression, of same
+            length as ``coeffs``
+
+    Keyword Args:
+        register (list): list of coordinates (in micrometers) of each atom in the ensemble
+        transition (list): information about the amplitude, phase, detuning and wires of the pulses
+        interaction_coeff (float): Rydberg interaction constant in units: :math:`MHz \times \mu m^6`.
+            Defaults to :math:`862690 \times 2\pi MHz \times \mu m^6`.
+
+    Returns:
+        RydbergHamiltonian: class representing the Hamiltonian of an ensemble of Rydberg atoms
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        coeffs,
+        observables,
+        register: list = None,
+        pulses: List["RydbergPulse"] = None,
+        interaction_coeff: float = 862690 * np.pi,
+    ):
+        self.register = register
+        self.pulses = [] if pulses is None else pulses
+        self.interaction_coeff = interaction_coeff
+        super().__init__(coeffs, observables)
+
+    def __add__(self, other):
+        if not isinstance(other, RydbergHamiltonian):
+            return super().__add__(other)
+
+        # Update coeffs, obs and hardware attributes
+        if self.register is not None:
+            if other.register is not None:
+                raise ValueError("We cannot add two Hamiltonians with an interaction term!")
+            if not other.wires.contains_wires(self.wires):
```

Comment:
This is the branch where the `register` is on `self`; it's `self.wires` that should contain the wires of `other`. Currently you get the warning for:

```
atom_coordinates = [[0, 0], [0, 5], [5, 0]]

Hd = rydberg_interaction(register=atom_coordinates, wires=[0, 1, 2])
Ht = rydberg_transition(2, 3, 4, wires=1)

Hd+Ht
```

```suggestion
            if not self.wires.contains_wires(other.wires):
```